addons/sale_loyalty/tests/test_program_rules.py

```diff
@@ -233,8 +233,19 @@ def test_program_rules_validity_dates_and_uses(self):
         self._auto_rewards(order, self.immediate_promotion_program)
         self.assertEqual(len(order.order_line.ids), 2, "The promo offer shouldn't have been applied we're not between the validity dates")
 
+        self.immediate_promotion_program.write({
+            'date_to': False,
+            'date_from': date.today() + timedelta(days=2),
+            'limit_usage': True,
+            'max_usage': 1,
```

Comment:
These lines are the same as the first created ones. They are not necessary when you update a record.